mmctl tool documentation

justinegeffen opened the pull request:
Documentation for mmctl tool.

wiersgallak commented:
Recommend bolding the 'options', 'format' and 'options inherited from parent commands'

wiersgallak commented:
When reviewing all the existing CLI to new MMCTL references, the ``mattermost channel modify`` to ``mmctl channel make-private`` seemed to be an outlier that didn't map as consistently as the others.  Wondering if we should omit until we have ``mmctl channel make-public`` to ensure we have full coverage of what this command provides in the mattermost CLI.

justinegeffen commented:
Both changes have been made as suggested, thank you!

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2320 @@
+==============================
+
+The mmctl tool is a remote CLI tool for Mattermost which is installed locally and uses the Mattermost API. Authentication
+is done with either login credentials or an authentication token. <elaborate more on this, provide links to other docs>.
```
There is an internal note here to elaborate / add links

justinegeffen replied:
Thanks for catching that!

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2320 @@
+**Notes**
+
+ -  Parameters in CLI commands are order-specific.
+ -  If special characters (``!``, ``|``, ``(``, ``)``, ``\``, ``'``, and ``"``) are used, the entire argument needs to be surrounded
```
Some of these bullet points need formatting tweaks.

justinegeffen replied:
Thanks! Fixed.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2320 @@
+
+
+Options
+.. code-block:: sh
```
This table might need formatting fixes.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+  -m, --mfa-token string      MFA token for the credentials
+  -p, --password string       Password for the credentials
+
+Options Inherited from Parent Commands
```
```suggestion
**Options Inherited from Parent Commands**
```

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+   --format string the format of the command output [plain, json] (default "plain")
+
+
+Authenticate to a server (e.g. >mmctl auth login https://test.mattermost.com). Then username and password
```
```suggestion
Authenticate to a server (e.g. >mmctl auth login https://test.mattermost.com), then username and password
```

Not sure if keeping the sentences separate is intentional.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+   $ mmctl auth login https://community.mattermost.com --name community --access-token MY_ACCESS_TOKEN
+
+
+into a Mattermost instance:
```
Could be intentional but this sentence seems incomplete.

justinegeffen replied:
Fixed.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+ --format string   the format of the command output [plain, json] (default "plain")
+
+
+mmctl channel make
```
These are marked as in progress. May be intentional.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+mmctl team archive
+^^^^^^^^^^^^^^^^
+
+*In progress*
```
In progress

justinegeffen replied:
Same as previous comment - I worked off the migration sheet and the repo and thought it might be helpful to list them all but mark which ones are still being worked on. Can definitely remove this.

justinegeffen replied:
Removed.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+   -h, --help       help for user
+
+
+mmctl user activate
```
Code blocks in "mmctl user activate" section are not rendering correctly.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2318 @@
+
+  --format string   the format of the command output [plain, json] (default "plain")
+
+mmctl version
```
Few sections here are incomplete. May be intentional

justinegeffen replied:
Removed.

amyblais commented on source/administration/command-line-tools.rst:
```diff
@@ -587,33 +644,12 @@ mattermost config migrate
 
        ./mattermost config migrate  path/to/config.json "postgres://mmuser:mostest@dockerhost:5432/mattermost_test?sslmode=disable&connect_timeout=10"
-       
-mattermost config reset
```
This would also need to be removed from the table of contents https://github.com/mattermost/docs/blob/e2b8f3b558288663b925d7ed3fa8970191c4f134/source/administration/command-line-tools.rst#mattermost-config.

justinegeffen replied:
I've re-added that command to the CLI docs - I must have deleted it inadvertently when I was trying to refine the formatting or copying content across. Thanks!

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2271 @@
+       --format string    the format of the command output [plain, json] (default "plain")
+       -h, --help         help for mmctl
+
+**Commands**
```
Missing 'mmctl auth'.

amyblais replied:
'version' and 'webhook' can be removed if we decide to remove from this doc for now.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2271 @@
+    -  `mmctl channel create`_ - Create a channel
+    -  `mmctl channel list`_ - List all channels on specified teams
+    -  `mmctl channel make_private`_ - Set a channel's type to "private"
+    -  `mmctl channel make`_ - Make a channel
```
'make', 'modify', 'move' commands are not in the documentation yet.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2271 @@
+
+Management of teams.
+
+Child Commands
```
'archive' and 'rename' can be removed.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2271 @@
+
+Management of users.
+
+Child Commands
```
'user activate' can be removed.

justinegeffen commented:
@wiersgallak, I've added the additional content we discussed on Friday. I cross-checked the migration list with what's in the repo and open PRs - can you confirm that the listed outstanding commands are correct?

I've also added the team users command to the docs.

ethervoid commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2346 @@
+Installing mmctl
+----------------
+
+To install the project in your `$GOPATH` run the following command as admin user:
```
We can include here the use of `brew install mmctl` command too. What do you think @mgdelacroix ?

justinegeffen replied:
This is a good suggestion. Would you suggest breaking this into sub-sections for different OSs or including it as an alternative in a single sentence?

ethervoid replied:
That would be great. We could do it by OS or we can have two sections one for building the tool and the other one to install from package manager

mgdelacroix replied:
I agree that separating the build and the install process would be good. I think that for the different install processes we can create a "Methods" section, and then describe ways to install it "via go get", "via brew", "via release package", etc. Chances are that we end up having more package managers to add here, and as some OSs share install methods (release and go get are for all of them, brew is for OSX and Linux...) I think having them under "Methods" suits it better

justinegeffen replied:
Great, thank you! Would you be open to sharing the methods and I can add them to the doc and do the restructuring?

mgdelacroix replied:
Sure, right now we have:

### Via brew

Valid for Linux and Mac, with brew installed, running `brew install mmctl` will install the tool.

### Via go get

Valid for Linux, Mac and Windows, with the go environment configured, running `go get -u github.com/mattermost/mmctl` would install the tool.

### Via release package

Valid for Linux, Mac and Windows, user needs to go to the [mmctl releases page](https://github.com/mattermost/mmctl/releases) and download the appropriate release for their OS, which will contain the binary of the tool.

justinegeffen replied:
I've updated the doc with this feedback.

amyblais commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -0,0 +1,2346 @@
+In the future, the Mattermost CLI tool will be deprecated.
+
+This feature was developed to a large extent by community contributions and we'd like to extend our gratitude to the contributors who have
+worked on this project. We are currently accepting PRs for Help Wanted issues
```
```suggestion
worked on this project. We are currently accepting pull requests for Help Wanted issues
```

amyblais replied:
This one is not fixed anymore.

justinegeffen replied:
That's my bad - I thought I'd committed it and must have over-written the change. Thanks!

mgdelacroix commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -51,12 +51,37 @@ the unit test coverage campaign for mmctl in the `Unit testing mmctl commands <h
 
 .. code-block:: sh
 
    go get -u github.com/mattermost/mmctl
```
This was covered in the "via go get" method, so we can safely remove it

mgdelacroix commented on source/administration/mmctl-cli-tool.rst:
```diff
@@ -51,12 +51,37 @@ the unit test coverage campaign for mmctl in the `Unit testing mmctl commands <h
+
+**Via go get (Linux, macOS, Windows)**
+
+Use this option on Linux, macOS, and Windows if you have a go environment configured.
```
The `$GOPATH` environment variable reference belongs to this method:

```suggestion
Use this option on Linux, macOS, and Windows if you have a go environment configured. This would install the project in your `$GOPATH`:
```

justinegeffen replied:
Changed it slightly differently. Let me know if it doesn't work.

mgdelacroix replied:
@justinegeffen just one nit, you don't need to be the admin user to install the tool, just to have the go environment set up.

justinegeffen replied:
Fixed, thank you!
